pkg/controller/controller_utils.go

```diff
@@ -124,9 +124,6 @@ func (r *RCExpectations) ExpectDeletions(rc *api.ReplicationController, dels int
 // Decrements the expectation counts of the given rc.
 func (r *RCExpectations) lowerExpectations(rc *api.ReplicationController, add, del int) {
 	if podExp, exists, err := r.GetExpectations(rc); err == nil && exists {
-		if podExp.add > 0 && podExp.del > 0 {
```

Comment:
Yeah the memory won't get deallocated. It's complaining because it's technically a race (imagine a system where 64 bit ints took two words).